### PR TITLE
material button implementation for openharmony

### PR DESCRIPTION
**Pull request by PatExplores**

### CODE REVIEW

File: Material_Button/package.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "@ohos/material_button",
```
Comment (archanaatr) [inline Material_Button/package.json:2]: change the name to material-button or materialbutton

File: Material_Button/src/main/ets/components/ToggleButtonModel.ets [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ToggleButtonOptions, ToggleSelection, ButtonVariant, ButtonOptions, TextualToggleType, IconToggleType } from "./Types"
+import { ButtonModel } from "./ButtonModel"
+
+export class ToggleButtonModel implements ToggleButtonOptions{
```
Comment (archanaatr) [inline Material_Button/src/main/ets/components/ToggleButtonModel.ets:20]: format all ets files

File: Material_Button/src/main/ets/components/ToggleButtonModel.ets [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  data: TextualToggleType[] | IconToggleType[] = []
+  width: number = 40
```
Comment (archanaatr) [inline Material_Button/src/main/ets/components/ToggleButtonModel.ets:22]: mark all dimentions as vp and fontsizes as fp

File: Material_Button/src/main/ets/components/ToggleButtonModel.ets [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+      this.buttonSelStatus.push(false)
+      if (entry.checked) {
+        this.checkOrUncheck(index)
+      }
+    })
+  }
+
+  update(options: ToggleButtonOptions) {
+    if (options) {
+      if (options.data != undefined) this.data = options.data
```
Comment (archanaatr) [inline Material_Button/src/main/ets/components/ToggleButtonModel.ets:90]: better to add null check also.

File: Material_Button/package.json (hunk 1 above)
Comment (PatExplores) [reply Material_Button/package.json:2]: name changed.

File: Material_Button/src/main/ets/components/ToggleButtonModel.ets (hunk 2 above)
Comment (PatExplores) [reply Material_Button/src/main/ets/components/ToggleButtonModel.ets:20]: formatted

File: Material_Button/src/main/ets/components/ToggleButtonModel.ets (hunk 4 above)
Comment (PatExplores) [reply Material_Button/src/main/ets/components/ToggleButtonModel.ets:90]: var x = null //Or unknown

if (x) {//fails for both null and unknown
    console.log("inside")
}
console.log("outside")

as this code handles both, we need not add explicitly.

**Comment by PatExplores** [comment]
require rework

**Pull request closed without merging**